packages/contracts/contracts/voting/admin-address/AdminAddressSetup.sol

```diff
@@ -0,0 +1,120 @@
+// SPDX-License-Identifier: MIT
+
+pragma solidity 0.8.10;
+
+import {Clones} from "@openzeppelin/contracts/proxy/Clones.sol";
+
+import {IDAO} from "../../core/IDAO.sol";
+import {DAO} from "../../core/DAO.sol";
+import {PermissionLib} from "../../core/permission/PermissionLib.sol";
+import {PluginSetup, IPluginSetup} from "../../plugin/PluginSetup.sol";
+import {AdminAddress} from "./AdminAddress.sol";
+
+/// @title AdminAddressSetup
+/// @author Aragon Association - 2022
+/// @notice The setup contract of the `AdminAddress` plugin.
+contract AdminAddressSetup is PluginSetup {
+    using Clones for address;
+
+    /// @notice The address of `AdminAddress` plugin logic contract to be cloned.
+    address private immutable adminAddressBase;
+
+    /// @notice The address zero to be used as oracle address for permissions.
+    address private constant NO_ORACLE = address(0);
+
+    /// @notice Thrown if admin address is zero.
+    /// @param admin The admin address.
+    error AdminAddressInvalid(address admin);
+
+    /// @notice The contract constructor, that deployes the `AdminAddress` plugin logic contract.
+    constructor() {
+        adminAddressBase = address(new AdminAddress());
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareInstallationDataABI() external pure returns (string memory) {
+        return "(address admin)";
+    }
+
+    /// @inheritdoc IPluginSetup
+    function prepareInstallation(address _dao, bytes memory _data)
+        external
+        returns (
+            address plugin,
+            address[] memory helpers,
+            PermissionLib.ItemMultiTarget[] memory permissions
+        )
+    {
+        IDAO dao = IDAO(_dao);
+
+        // Decode `_data` to extract the params needed for cloning and initializing `AdminAddress` plugin.
+        address admin = abi.decode(_data, (address));
+
+        if (admin == address(0)) {
+            revert AdminAddressInvalid({admin: admin});
+        }
+
+        // Clone plugin contract.
+        plugin = adminAddressBase.clone();
+
+        // Initialize cloned plugin contract.
+        AdminAddress(plugin).initialize(dao);
+
+        // Prepare helpers
+        helpers = new address[](0);
```

Comment:
I think this line can be ommitted

Reply:
it can not,
it will generate warning
i prefer no warning

Reply:
@Rekard0  do the following:

(helpers); // silence the warning.